In [1]:
import pandas as pd
import numpy as np 

In [2]:
path = '.\\data\\undergrad_terms.xlsx'
df = pd.read_excel(path, header=0)  

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 727369 entries, 0 to 727368
Data columns (total 13 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   id                   727369 non-null  object 
 1   institution          727369 non-null  object 
 2   cip6                 727369 non-null  object 
 3   term                 727369 non-null  int64  
 4   level                727369 non-null  object 
 5   standing             727369 non-null  object 
 6   coop                 727369 non-null  object 
 7   hours_term_attempt   727369 non-null  float64
 8   hours_term           727369 non-null  float64
 9   gpa_term             727102 non-null  float64
 10  hours_cumul_attempt  719274 non-null  float64
 11  hours_cumul          727369 non-null  float64
 12  gpa_cumul            727156 non-null  float64
dtypes: float64(6), int64(1), object(6)
memory usage: 72.1+ MB


1. How many observations and how many variables are there in the dataset?

There are 727368 observations and 13 variables in the dataset.


In [13]:
df = df.astype({'term': 'object'})

2. What is the dtype of the variable "term"? Please change the dtype to "object".

The dtype of the variable "term" is int64.

In [12]:
df['level'].unique()  

array(['01 Freshman', '02 Sophomore', '03 Junior', '04 Senior'],
      dtype=object)

In [11]:
df['standing'].unique()

array(['First Time in College', 'Good Standing', 'Academic Probation',
       'New Transfer', 'Returning',
       'Transfer from Florida Community College', 'Academic Notice',
       'Suspended'], dtype=object)

3. What are the categories of the variables "level" and "standing"? 

'level' = '01 Freshman', '02 Sophomore', '03 Junior', '04 Senior'

'standing' = 'First Time in College', 'Good Standing', 'Academic Probation',
       'New Transfer', 'Returning',
       'Transfer from Florida Community College', 'Academic Notice',
       'Suspended'

In [3]:
pd.isna(df)

id  institution   cip6   term  level  standing   coop  \
0       False        False  False  False  False     False  False   
1       False        False  False  False  False     False  False   
2       False        False  False  False  False     False  False   
3       False        False  False  False  False     False  False   
4       False        False  False  False  False     False  False   
...       ...          ...    ...    ...    ...       ...    ...   
727364  False        False  False  False  False     False  False   
727365  False        False  False  False  False     False  False   
727366  False        False  False  False  False     False  False   
727367  False        False  False  False  False     False  False   
727368  False        False  False  False  False     False  False   

        hours_term_attempt  hours_term  gpa_term  hours_cumul_attempt  \
0                    False       False     False                False   
1                    False       False     False                False   
2                    False       False     False                False   
3                    False       False     False                False   
4                    False       False     False                False   
...                    ...         ...       ...                  ...   
727364               False       False     False                False   
727365               False       False     False                False   
727366               False       False     False                False   
727367               False       False     False                False   
727368               False       False     False                False   

        hours_cumul  gpa_cumul  
0             False      False  
1             False      False  
2             False      False  
3             False      False  
4             False      False  
...             ...        ...  
727364        False      False  
727365        False      False  
727366        False      False  
727367        False      False  
727368        False      False  

[727369 rows x 13 columns]

In [20]:
df.describe() 

hours_term_attempt     hours_term       gpa_term  hours_cumul_attempt  \
count       727369.000000  727369.000000  727102.000000        719274.000000   
mean            12.569747      11.483643       2.862365            71.838891   
std              4.177008       4.544693       0.929180            58.703793   
min              0.000000       0.000000       0.000000             0.000000   
25%             12.000000       9.000000       2.400000            32.000000   
50%             13.000000      12.000000       3.000000            64.000000   
75%             15.000000      15.000000       3.540000           100.000000   
max             64.000000      46.000000       9.300000          2169.000000   

         hours_cumul      gpa_cumul  
count  727369.000000  727156.000000  
mean       70.699505       2.796077  
std        47.308674       0.780409  
min         0.000000       0.000000  
25%        32.000000       2.390000  
50%        64.000000       2.900000  
75%       100.000000       3.350000  
max       546.000000       4.000000

4. How many missing values are there in the dataset (for each variable)? Please also provide a summary of descriptive statistics of all numerical variables. 

There are no missing values in the dataset.



In [6]:
df.groupby('institution')[['id','gpa_term']].describe() 

gpa_term                                                
                  count      mean       std  min   25%   50%   75%  max
institution                                                            
Institution A   30533.0  2.425978  1.086197  0.0  1.85  2.60  3.21  4.0
Institution B   95548.0  2.923271  0.789048  0.0  2.50  3.04  3.50  4.0
Institution C   79214.0  2.888978  0.844113  0.0  2.44  3.00  3.51  9.3
Institution D    4110.0  3.201876  0.567747  0.9  2.87  3.30  3.63  4.0
Institution E   66799.0  2.848542  0.962117  0.0  2.40  3.00  3.60  4.0
Institution F   48763.0  2.567827  1.157379  0.0  2.00  2.83  3.42  4.0
Institution G   33602.0  2.505875  1.101988  0.0  2.00  2.69  3.27  4.0
Institution H   70985.0  2.835634  0.856296  0.0  2.31  3.00  3.50  4.0
Institution J  105133.0  2.961747  0.846880  0.0  2.56  3.11  3.58  4.0
Institution K  105571.0  2.985048  0.939130  0.0  2.60  3.20  3.70  4.0
Institution L   25984.0  3.156223  0.831707  0.0  2.75  3.33  3.80  4.0
Institution M   60860.0  2.897324  0.886141  0.0  2.40  3.00  3.57  4.0

In [9]:
print(df['gpa_term'].mean())
print(df['gpa_term'].max())

2.8623647713801974
9.3


5. Please limit the dataset to include only two columns: `id` and `gpa_term`. Then group the data by `id`, obtain the average `gpa_term` and the maximum `gpa_term` for each student. 

Mean: 2.8623647713801974

Max: 9.3